app/js/components/Buyer/OfferTrade.jsx

```diff
@@ -1,40 +1,58 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import {withNamespaces} from "react-i18next";
 import {Row, Col, FormGroup, Button} from 'reactstrap';
 import Form from 'react-validation/build/form';
 import Input from 'react-validation/build/input';
 import Blockies from 'react-blockies';
-import {isNumber} from "../../validators";
+import {isNumber, lowerEqThan, higherEqThan} from "../../validators";
 
-const OfferTrade = ({address, name, min, max, fiat, asset}) => (
+const OfferTrade = ({address, name, min, max, fiat, asset, onClick, assetQty, fiatQty, onFiatChange, onAssetChange, disabled, t}) => (
   <Row>
```

Comment:
I think this component can be split

Reply:
I think I could split the Form out, but it makes it so that we have a lot of props to pass and just makes the app heavier in terms of files. Do you think it's really worth it? Especially since I don't think we will reuse that component.

Reply:
The idea would be to pass less props, for example in this one I think I can see 2 components being called from the container:

```
<Col xs="12" className="mt-5 text-center">
      <h2>Trade amount with <br/><span><Blockies seed={address || '0x0'} className="rounded-circle"/> {name}</span></h2>
      <p className="mt-3">Min: {min}{fiat.symbol} - Max: {max}{fiat.symbol}</p>
    </Col>
```
and
```
<Col xs="12" className="mt-4">
      <Form className="text-center">
        <FormGroup>
          <Input type="number" name="fiat" className="form-control" value={fiatQty}
                 validations={[isNumber, lowerEqThan.bind(null, max), higherEqThan.bind(null, min)]}
                 placeholder="Fiat quantity" onChange={(e) => onFiatChange(e.target.value)}/>
          <span className="input-icon">{fiat.id}</span>
        </FormGroup>
        <FormGroup>
          <Input type="number" name="asset" className="form-control" value={assetQty} validations={[isNumber]}
                 placeholder="Asset quantity" onChange={(e) => onAssetChange(e.target.value)}/>
          <span className="input-icon">{asset}</span>
        </FormGroup>
        {disabled && <p className="text-info">{t('buyer.offerTrade.enterBefore')}</p>}
        <Button color="primary" className="px-4" onClick={onClick} disabled={disabled}>Open trade</Button>
      </Form>
    </Col>
```

If you think it is not worth it, it is fine, up to you!